components/brave_wallet/browser/brave_wallet_service.cc

```diff
@@ -512,6 +512,11 @@ void BraveWalletService::SetDefaultBaseCryptocurrency(
   }
 }
 
+void BraveWalletService::GetShowWalletTestNetworks(
```

Comment:
Please add a test for this method being backed by a pref.

Reply:
Test added!
https://github.com/brave/brave-core/pull/13098/files#diff-c114760829ee3614f7d0bcef91204dc45e57455128412f7b4f66704df7c24bc6R1084-R1090

Reply:
I meant it should cover case when changing pref changes service method outcome;)

<details>
  <summary>So please add smth like this</summary>
  
```
diff --git a/browser/brave_wallet/brave_wallet_service_unittest.cc b/browser/brave_wallet/brave_wallet_service_unittest.cc
index 93c5311959..7e686898b5 100644
--- a/browser/brave_wallet/brave_wallet_service_unittest.cc
+++ b/browser/brave_wallet/brave_wallet_service_unittest.cc
@@ -459,6 +459,18 @@ class BraveWalletServiceUnitTest : public testing::Test {
     return default_cryptocurrency;
   }
 
+  bool GetShowWalletTestNetworks() {
+    base::RunLoop run_loop;
+    bool show_wallet_test_networks;
+    service_->GetShowWalletTestNetworks(
+        base::BindLambdaForTesting([&](bool b) {
+          show_wallet_test_networks = b;
+          run_loop.Quit();
+        }));
+    run_loop.Run();
+    return show_wallet_test_networks;
+  }
+
   void SimulateOnGetImportInfo(const std::string& new_password,
                                bool result,
                                const ImportInfo& info,
@@ -1084,7 +1096,10 @@ TEST_F(BraveWalletServiceUnitTest, GetAndSetDefaultBaseCryptocurrency) {
 TEST_F(BraveWalletServiceUnitTest, GetShowWalletTestNetworks) {
   // Default value for kShowWalletTestNetworks should be false
   EXPECT_FALSE(GetPrefs()->GetBoolean(kShowWalletTestNetworks));
-  EXPECT_FALSE(GetShowWalletTestNetworks(GetPrefs()));
+  EXPECT_FALSE(GetShowWalletTestNetworks());
+
+  GetPrefs()->SetBoolean(kShowWalletTestNetworks, true);
+  EXPECT_TRUE(GetShowWalletTestNetworks());
 }
 
 TEST_F(BraveWalletServiceUnitTest, EthAddRemoveSetUserAssetVisible) {

```
  
</details>

Marking as approved.

Reply:
Gotcha!
Updated, thank you for the help! 😎